packages/components/checkbox/tests/use-checkbox.test.tsx

```diff
@@ -0,0 +1,267 @@
+    fireEvent.touchStart(InputBoxComponent)
+    expect(InputBoxComponent).toBeInTheDocument()
+  })
+
+  test("should call default onMouseDown on icon props when no handler is passed", () => {
+    const InputBox = () => {
+      const { getIconProps } = useCheckbox({})
+      const iconProps = { ...getIconProps() }
+
+      return <input type="checkbox" data-testid="input-box" {...iconProps} />
+    }
+
+    render(<InputBox />)
+    const InputBoxComponent = screen.getByTestId("input-box")
+    fireEvent.mouseDown(InputBoxComponent)
+    expect(InputBoxComponent).toBeInTheDocument()
+  })
+
+  test("should call default onChange on input props when no handler is passed", () => {
+    const InputBox = () => {
+      const { getInputProps } = useCheckbox({})
+      const inputProps = { ...getInputProps() }
+
+      return <input type="checkbox" data-testid="input-box" {...inputProps} />
+    }
+
+    render(<InputBox />)
+    const InputBoxComponent = screen.getByTestId("input-box")
+    fireEvent.change(InputBoxComponent)
+    expect(InputBoxComponent).toBeInTheDocument()
+  })
+
+  test("should call default onKeyDown on input props when no handler is passed", () => {
+    const InputBox = () => {
+      const { getInputProps } = useCheckbox({})
+      const inputProps = { ...getInputProps() }
+
+      return <input type="checkbox" data-testid="input-box" {...inputProps} />
+    }
+
+    render(<InputBox />)
+    const InputBoxComponent = screen.getByTestId("input-box")
+    fireEvent.keyDown(InputBoxComponent)
+    expect(InputBoxComponent).toBeInTheDocument()
+  })
+
+  test("should call default onKeyUp on input props when no handler is passed", () => {
+    const InputBox = () => {
+      const { getInputProps } = useCheckbox({})
+      const inputProps = { ...getInputProps() }
+
+      return <input type="checkbox" data-testid="input-box" {...inputProps} />
+    }
+
+    render(<InputBox />)
+    const InputBoxComponent = screen.getByTestId("input-box")
+    fireEvent.keyUp(InputBoxComponent)
+    expect(InputBoxComponent).toBeInTheDocument()
+  })
+
+  test("should call default click on container props when no handler is passed", () => {
+    const InputBox = () => {
+      const { getContainerProps } = useCheckbox({})
+      const containerProps = { ...getContainerProps() }
+
+      return (
+        <label {...containerProps} data-testid="input-box-container">
+          <input type="checkbox" />
+        </label>
+      )
+    }
+
+    render(<InputBox />)
+    const InputBoxContainerComponent = screen.getByTestId("input-box-container")
+    fireEvent.click(InputBoxContainerComponent)
+    expect(InputBoxContainerComponent).toBeInTheDocument()
+  })
+
+  test("Does disabled function properly", () => {
+    const { rerender, result } = renderHook(() =>
+      useCheckbox({ disabled: false }),
+    )
+    expect(result.current.focused).toBeFalsy()
+
+    rerender({ disabled: true })
+    expect(result.current.focused).toBeFalsy()
+  })
```

Comment:
This test is unnecessary because it cannot detect that `disabled` is not working properly.